packages/common/e2e/src/utils/versions.ts

```diff
@@ -1,7 +1,22 @@
 import { readJsonFile } from '@nx/devkit';
 
-export function getNxVersion(): string {
+// eslint-disable-next-line unicorn/import-style
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+
+async function getLatestNxVersion(): Promise<string> {
+    const { stdout, stderr } = await exec('npm view nx version');
+    return stderr || stdout;
+}
```

Comment:
Two approaches to this, we either rely on the code we have here to determine the latest nx version, or we set the existing `NX_VERSION` variable in the new workflow that has been created. 

This could be retrieved through the following action: https://github.com/marketplace/actions/package-latest-version
You could then access the output of this package and assign it to `NX_VERSION` within an "env" section of the e2e job.

What do you think?

Reply:
I took a look at this. I didn't seem to be working properly, but maybe I didn't have the workflow file setup correctly. I think we can leave this for now to avoid this ticket dragging on, and look to implement something like this in the future if we think its necessary.